pyccel/ast/builtin_methods/set_methods.py

```diff
@@ -32,20 +32,15 @@ class SetAdd(PyccelInternalFunction) :
     """
     __slots__ = ("_set_variable", "_add_arg")
     _attribute_nodes = ("_set_variable", "_add_arg")
-    _dtype = NativeVoid()
     _shape = None
     _order = None
     _rank = 0
-    _precision = None
-    _class_type = NativeVoid()
+    _class_type = VoidType()
     name = 'add'
 
     def __init__(self, set_variable, new_elem) -> None:
         is_homogeneous = (
-            new_elem.dtype is not NativeGeneric() and
-            set_variable.dtype is not NativeGeneric() and
             set_variable.dtype == new_elem.dtype and
```

Comment:
I think this line should also be deleted and replace it with the following line : 
```suggestion
         self._class_type = set_variable.class_type.element_type
```

Reply:
I agree that this line is not correct but I'm not sure I agree with your replacement. Shouldn't it be:
```suggestion
            set_variable.class_type == new_elem.class_type and
```
?

Reply:
a137c5fd

Reply:
I'm still unsure why we're not checking for `element_type`.

Reply:
:facepalm: Your are correct. Fixed: 690fa985

Reply:
This is still not what you suggested though. Do you agree that it should be a test not an assignment?